ksqldb-functional-tests/src/test/resources/query-validation-tests/partition-by.json

```diff
@@ -1050,6 +1050,30 @@
         ]
       }
     },
+    {
+      "name": "multiple columns - select star - reorder columns",
+      "statements": [
+        "CREATE STREAM INPUT (NAME STRING, ID INT, AGE INT) with (kafka_topic='input', format='JSON');",
+        "CREATE STREAM OUTPUT AS select * from INPUT partition by AGE, ID;"
```

Comment:
meta question: The lower/upper case style is confusing to me.

Should we not upper case all keywords, and lower case all identifiers? This is all a wild mix making it hard to read.

```
CREATE STREAM input (name STRING, id INT, age INT) with (kafka_topic='input', format='JSON');
CREATE STREAM output AS SELECT * FROM input PARTITION BY age, id;
```

Reply:
Can we replicate this test (or can you point out existing tests) for:
 - input stream with existing multi-column key plus reordering: `PARTITION BY key2,key1`
 - `PARTITION BY key, nonkey`
 - `PARTITION BY nonKey, key`
 - all of the above replicated `SELECT columnNames` (instead of `*`) with all 4 reordering combination (either in SELECT or PARTITION BY); seems two are already covered
 - mixing key and values in select: `SELECT key2, nonKey2, key1, nonKey1...`

Reply:
* `"name": "multiple key columns - reordered",` already exists
* `"name": "multiple columns - some key some value"` already exists
* I'll add that
* `multiple key columns` tests this
* the one I mentioned above tests this as well

As far as the casing style I just copied existing tests - happy to change the new ones but I'm not going to go and change the old ones for this PR

Reply:
> 'm not going to go and change the old ones for this PR

Sure :) (did not expect that)